spatial/r2/vector.go

```diff
@@ -54,6 +54,24 @@ func Norm2(p Vec) float64 {
 	return p.X*p.X + p.Y*p.Y
 }
 
+// Unit returns the unit vector colinear to p.
+func Unit(p Vec) Vec {
+	if p.X == 0 && p.Y == 0 {
+		return Vec{}
+	}
+	return p.Scale(1 / Norm(p))
+}
+
+// CosTheta returns the cosine of the opening angle between p and q.
+func CosTheta(p, q Vec) float64 {
+	var (
+		np2 = Norm2(p)
+		nq2 = Norm2(q)
+		dot = p.Dot(q)
+	)
+	return dot / math.Sqrt(np2*nq2)
```

Comment:
`return p.Dot(q) / (Norm(p)*Norm(q))`? Slower maybe, but more accurate.

Reply:
had to actually introduce a tolerance (`1e-14`) with this.